src/main/java/pl/betoncraft/betonquest/conditions/DateCondition.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * BetonQuest - advanced quests for Bukkit
+ * Copyright (C) 2016  Jakub "Co0sh" Sapalski
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package pl.betoncraft.betonquest.conditions;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+
+import pl.betoncraft.betonquest.Instruction;
+import pl.betoncraft.betonquest.InstructionParseException;
+import pl.betoncraft.betonquest.QuestRuntimeException;
+import pl.betoncraft.betonquest.api.Condition;
+
+/**
+ * Returns true if given date is matched
+ *
+ * @author Archerymaister
+ */
+public class DateCondition extends Condition {
+
+	private Date startDate;
+	private Date endDate;
+	private SimpleDateFormat dateFormat;
+
+	public DateCondition(Instruction instruction) throws InstructionParseException {
+		super(instruction);
+		super.staticness = true;
+		super.persistent = true;
+
+		if(instruction.size() > 3){
+			throw new InstructionParseException("Too many arguments!");
+		}
+		if(instruction.size() < 2){
+			throw new InstructionParseException("not enough arguments!");
+		}
+		if (instruction.size() == 2) {
```

Comment:
From reading your code I assume that if there is only one date, the condition will be true that whole day. It's nice to be able to achieve that with a single argument, but it complicates the code a lot. Why don't you leave it to require always two arguments and let the users type it like `14.01.2018-00:00:00 15.01.2018-00:00:00`? It will be more explicit and the code will be easier to reason about.

Just a suggestion though.